### PR TITLE
feat: improve token list

### LINKED ISSUE #3736: [Bug] MASK icon wasn't showed in tradeview swap's select box since MASK is auto filled in BSC network.
### How to reproduce?

1.Change wallet's network to BSC;
2.search #mask and check swap section of Tradeview;

### What happened?
MASK icon wasn't showed in select box but showed in token list.
截屏2021-07-22 下午5 36 48

### What should happen?
MASK icon should be showed in both select box and token list.

### Mask Version:

- [x] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #mask-129
closes #3736

### CODE REVIEW

**Review by UncleBill — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

File: packages/web3-shared/src/utils/token.ts [hunk 1]
```diff
@@ -145,26 +145,29 @@ const MASK_ADDRESS = getTokenConstants().MASK_ADDRESS
 export const getTokenUSDValue = (token: Asset) => (token.value ? Number.parseFloat(token.value[CurrencyType.USD]) : 0)
 export const getBalanceValue = (asset: Asset) => parseFloat(formatBalance(asset.balance, asset.token.decimals))
 
-export const sortToken = (a: FungibleTokenDetailed, b: FungibleTokenDetailed) => {
+export const sortToken = (a: FungibleTokenDetailed, b: FungibleTokenDetailed, isMaskBoost: boolean = false) => {
```
Comment (septs) [inline packages/web3-shared/src/utils/token.ts:148]:
```suggestion
export const makeSortTokenFn = (isMaskBoost = false) => (a: FungibleTokenDetailed, b: FungibleTokenDetailed) => {
```

File: packages/web3-shared/src/utils/token.ts [hunk 2]
```diff
@@ -145,26 +145,29 @@ const MASK_ADDRESS = getTokenConstants().MASK_ADDRESS
...
+    if (isMaskBoost) {
+        if (isSameAddress(a.address, MASK_ADDRESS ?? '')) return -1
+        if (isSameAddress(b.address, MASK_ADDRESS ?? '')) return 1
+    }
+
     return 0
 }
 
-export const sortAssert = (a: Asset, b: Asset, chainId: ChainId) => {
+export const sortAssert = (a: Asset, b: Asset, chainId: ChainId, isMaskBoost: boolean = false) => {
```
Comment (septs) [inline packages/web3-shared/src/utils/token.ts:162]:
```suggestion
export const makeSortAssert (chainId: ChainId, isMaskBoost = false) => (a: Asset, b: Asset) => {
```

Comment (septs) [reply packages/web3-shared/src/utils/token.ts:162]:
```js
dataset.sort(makeSortAssertFn(ChainId.Mainnet, true))
```

File: packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts [hunk 3]
```diff
@@ -97,11 +97,14 @@ async function fetchCommonERC20TokensFromTokenList(
  * @param chainId
  */
 function fetchERC20TokensFromTokenList(urls: string[], chainId = ChainId.Mainnet) {
-    return urls.map((x) => {
+    return urls.map(async (x) => {
```
Comment (septs) [inline packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:100]:
```suggestion
    return Promise.all(urls.map(async (x) => {
```

File: packages/maskbook/src/plugins/Trader/SNSAdaptor/trader/TradeForm.tsx [hunk 4]
```diff
@@ -115,6 +124,23 @@ export function TradeForm(props: TradeFormProps) {
...
+    // Work around the issue: https://github.com/DimensionDev/Maskbook/issues/3736
+    // Will be removed after Debank update mask token on BSC
+    const fixMaskToken = (token?: FungibleTokenDetailed): FungibleTokenDetailed | undefined => {
+        if (!token) return token
+        if (isSameAddress(token.address, MASK_ADDRESS)) {
+            return {
+                ...token,
+                ...{
+                    logoURI: [
+                        'https://raw.githubusercontent.com/DimensionDev/Maskbook-Website/master/img/MB--CircleCanvas--WhiteOverBlue.svg',
```
Comment (septs) [inline packages/maskbook/src/plugins/Trader/SNSAdaptor/trader/TradeForm.tsx:136]: plz add the file to maskbook-vi

File: packages/web3-shared/src/utils/token.ts [hunk 5]
```diff
@@ -136,4 +139,53 @@ export function parseStringOrBytes32(
         : defaultValue
 }
 
+//#region asset sort
+const MASK_ADDRESS = getTokenConstants().MASK_ADDRESS
```
Comment (septs) [inline packages/web3-shared/src/utils/token.ts:143]:
```suggestion
const { MASK_ADDRESS } = getTokenConstants()
```

File: packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts (hunk 3 above)
Comment (septs) [reply packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:100]: ?

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**